src/EditorFeatures/CSharp/CodeGeneration/CSharpCodeGenerationOptionsStorage.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+
+internal static class CSharpCodeGenerationOptionsStorage
+{
+    [ExportLanguageService(typeof(ICodeGenerationOptionsStorage), LanguageNames.CSharp), Shared]
+    private sealed class Service : ICodeGenerationOptionsStorage
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public Service()
+        {
+        }
+
+        public CodeGenerationOptions GetOptions(IGlobalOptionService globalOptions)
+            => GetCSharpCodeGenerationOptions(globalOptions);
```

Comment:
aside, why not pull in IGlobalOptionService  in teh importing constructor here.  tehn, the client of ICodeGenerationOptionsStorage doesn't need to get IGlobalOptionService themselves.  they just call .GetOptions() without having to care bout the impl detail of how that works.

Reply:
That would be possible, but the call site is an extension method on `IGlobalOptionService`:

```C#
    public static CodeGenerationOptions GetCodeGenerationOptions(this IGlobalOptionService globalOptions, HostLanguageServices languageServices)
        => languageServices.GetRequiredService<ICodeGenerationOptionsStorage>().GetOptions(globalOptions);
```

The caller already has the global options and asks for various options from it. So the `ICodeGenerationOptionsStorage` is the implementation detail here needed only to instantiate language specific options object.